README.md

```diff
@@ -1,196 +1,289 @@
+We recommend that the token given is limited in scope to the upload project, and has no higher than **CONTRIBUTE** privileges.
+#### Example Script
+The following is an example script that writes a flat file to the RUN directory once a RUN directory has been successfully streamed.
+Recall that the script will be triggered with a single command line parameter, where `$1` is the path to the local RUN directory  that has been successfully streamed to DNAnexus.
+```
+#!/bin/bash
+set -e -x -o pipefail
+rundir="$1"
+echo "Completed streaming run directory: $rundir" > "$rundir/COMPLETE.txt"
+```
+#### Actions performed by Role
+The dx-streaming-upload role perform, broadly, the following:
+1. Installs the DNAnexus tools [dx-toolkit](https://wiki.dnanexus.com/Downloads#DNAnexus-Platform-SDK) and [upload agent](https://wiki.dnanexus.com/Downloads#Upload-Agent) on the local machine.
+2. Set up a CRON job that monitors a given directory for RUN directories periodically, and streams the RUN directory into a DNAnexus project, triggering an app(let)/workflow upon successful upload of the directory and a local script (when specified by user)
+#### Downstream analysis
+The dx-streaming-upload role can optionally trigger a DNAnexus applet/workflow upon completion of incremental upload. The desired DNAnexus applet or workflow can be specified (at a per `monitored_user` basis) using the Ansible variables `applet` or `workflow` respectively (mutually exclusive, see explanantion of variables for general explanations).
+More information about DNAnexus workflows can be found at the [DNAnexus wiki page](https://wiki.dnanexus.com/API-Specification-v1.0.0/Running-Analyses)
+### Authorization
+The downstream analysis (applet or workflow) will be launched in the project into which the RUN directory is uploaded to (`project`). The DNAnexus user / associated `dx_token` or `dx_user_token` must have at least `CONTRIBUTE` access to the aforementioned project for the analysis to be launched successfully. Computational resources are billable and will be billed to the bill-to of the corresponding project.
+### Input and Options
+The specified applet/workflow will be triggered using the `run` [API](http://autodoc.dnanexus.com/bindings/python/current/dxpy_apps.html?highlight=applet%20run#dxpy.bindings.dxapplet.DXExecutable.run) in the dxpy tool suite.
+For an applet, the `executable_input` hash to the `run` command will be prepopulated with the key-value pair {"`upload_sentinel_record`": `$record_id`} where `$record_id` is the DNAnexus file-id of the sentinel record generated for the uploaded RUN directory (see section titled **Files generated**).
+For a workflow the `executable_input` hash will be prepoluated with the key-value pair {"`0.upload_sentinel_record`": `$record_id`} where `$record_id` is the DNAnexus file-id of the sentinel record generated for the uploaded RUN directory (see section titled **Files generated**).

```

Comment:
--> " For a workflow, the `executable_input` hash will be prepopulated with the key-value pair {"`0.upload_sentinel_record`": `$record_id`}, where `$record_id` is the DNAnexus file ID of the sentinel record generated for the uploaded RUN directory (see section titled **Files generated**). "

Reply:
Also: other references to other sections of the document use italics, while this one uses bold. I'm going to actually suggest that you use `monospace` or link to the sections appropriately.